.github/workflows/backport.yaml

```diff
@@ -25,7 +25,26 @@ jobs:
           git config user.name "github-actions[bot]"
           git config user.email "github-actions[bot]@users.noreply.github.com"
 
+      - name: Check if backports already exist
+        id: check-existing
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+        run: |
+          # Check for existing backport PRs for this PR number
+          EXISTING_BACKPORTS=$(gh pr list --state all --search "backport-${PR_NUMBER}-to" --json title,headRefName,baseRefName | jq -r '.[].headRefName')
+          
+          if [ -n "$EXISTING_BACKPORTS" ]; then
+            echo "Found existing backport PRs:"
+            echo "$EXISTING_BACKPORTS"
+            echo "skip=true" >> $GITHUB_OUTPUT
+            echo "::notice::Backport PRs already exist for PR #${PR_NUMBER}, skipping to avoid duplicates"
+          else
+            echo "skip=false" >> $GITHUB_OUTPUT
+          fi
+
       - name: Extract version labels
+        if: steps.check-existing.outputs.skip != 'true'
```

Comment:
Not for this PR at all, but when a bunch of steps have the same gate applied it makes me think they should be a single [composite action](https://docs.github.com/en/actions/tutorials/create-actions/create-a-composite-action#composite-actions-and-reusable-workflows)